byron/semantics/executable-spec/src/Control/State/Transition/Trace.hs

```diff
@@ -205,6 +207,27 @@ preStatesAndSignals OldestFirst tr
 preStatesAndSignals NewestFirst tr
   = reverse $ preStatesAndSignals OldestFirst tr
 
+-- | Apply the signals in the list and elaborate a trace with the resulting
+-- states.
+--
+-- If any of the signals cannot be applied, then it is discarded, and the next
+-- signal is tried.
+closure
+  :: forall s
+   . STS s
+   => Environment s
+   -> State s
+   -> [Signal s]
+   -> Trace s
+closure env st0 sigs = mkTrace env st0 $ loop st0 sigs []
```

Comment:
Seems an odd name for this. What's the motivation behind it?

Reply:
Yeah, now that you mention this I don't know whether it is the most appropriate name. It was the best I could come up with based on what this function does with its parameters:

```
trans sign-1 (trans sign-2 ( ... (trans sig0 st0) ...))
```

So I thought it can be seen as the transitive closure of the signals in `sigs`. Any suggestions for a better name?